index.js

```diff
@@ -1,20 +1,19 @@
 /* -*- coding: UTF-8, tab-width: 2 -*- */
 /*jslint indent: 2, maxlen: 80, node: true */
-'use strict';
-
-module.exports = function flatten(list, depth) {
-  depth = (typeof depth === 'number') ? depth : Infinity;
-
-  function coreFlatten(list, d) {
-    return list.reduce(function (acc, item) {
-      if (Array.isArray(item) && d < depth) {
-        return acc.concat(coreFlatten(item, d + 1));
-      }
-      return acc.concat(item);
-    }, []);
+"use strict";
+function flatRecurs(arr, d, res) {
+  var i = 0, cur, len = arr.length;
+  for (i; i < len; i += 1) {
+    cur = arr[i];
+    if (Array.isArray(cur) && d) {
```

Comment:
What's the `&& d` for? Is there any false-y value that would be accepted by Array.isArray?

Reply:
Oh, sorry. I didn't read properly. Nevermind.

Reply:
Then again, could it give better performance to switch them? Checking a boolean value should be cheaper than calling a function, but then again it depends on which of both checks will fail more often, and that depends on input.

Reply:
I've swapped them in any case. Going to close this PR and open a new one into experimental.